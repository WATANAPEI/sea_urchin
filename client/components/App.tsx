import React from "react";
// @ts-ignore
import HeaderBar from "./HeaderBar.tsx";
// @ts-ignore
import MainContainer from "../container/MainContainer.tsx";
// @ts-ignore
import SideBar from "./SideBar.tsx";
import Grid from "@material-ui/core/Grid";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
// @ts-ignore
import WordListContainer from "../container/WordListContainer.tsx";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    headerBar: {
      height: "20vh"
    },
    sideBar: {
      height: "80vh"
    },
    main: {
      height: "80vh"
    }
  })
);

function App() {
  const classes = useStyles();
  let backendUrl: string;
  const initialWord = {
    id: -1,
    word: "Initialize error",
    meaning: "Initialize error",
    wordLanguageID: -1,
    meaningLanguageID: -1
  };

  //console.log(`NODE_ENV: ${process.env.NODE_ENV}`);
  if (process.env.NODE_ENV === "production") {
    backendUrl = "https://wpei.dev/lang/api/v1";
  } else {
    backendUrl = "http://127.0.0.1:3000";
  }
  return (
    <Router>
      <Grid container spacing={2} className={classes.headerBar}>
        <Grid item id="headerBar" xs={12}>
          <HeaderBar text="Lang Project" />
        </Grid>
        <Grid item id="sideBar" xs={3} className={classes.sideBar}>
          <SideBar />
        </Grid>
        <Grid item xs={9} className={classes.main}>
          <Route id="main" exact path="/lang/words/" render={() => <MainContainer backendUrl={backendUrl} {...initialWord} />} />
          <Route id="wordList" path="/lang/words/wordlist" render={() => <WordListContainer backendUrl={backendUrl} {...initialWord} />} />
        </Grid>
      </Grid>
    </Router>
  );
}

export default App;
