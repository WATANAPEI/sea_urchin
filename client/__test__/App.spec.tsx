//jest.mock("userEnv");
//const apiUrl = jest.requireActual("../../.env/dev.ts");
import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
// @ts-ignore
import App from "../components/App.tsx";

describe("<App />", () => {
  const wrapper = createShallow()(<App />);
  it("has HeaderBar with a title text", () => {
    expect(wrapper.find("HeaderBar").text()).toBeTruthy();
  });
  it("has SideBar with a text", () => {
    expect(wrapper.find("SideBar").text()).toBeTruthy();
  });
  it("has Main component", () => {
    expect(wrapper.find("#main").exists()).toEqual(true);
  });
  it("has WordList component", () => {
    expect(wrapper.find("#wordList").exists()).toEqual(true);
  });
  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
