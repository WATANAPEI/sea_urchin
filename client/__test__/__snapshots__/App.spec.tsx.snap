// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders properly 1`] = `
<BrowserRouter>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-headerBar-1"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      id="headerBar"
      item={true}
      xs={12}
    >
      <HeaderBar
        text="Lang Project"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-sideBar-2"
      id="sideBar"
      item={true}
      xs={3}
    >
      <SideBar />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-main-3"
      item={true}
      xs={9}
    >
      <Route
        exact={true}
        id="main"
        path="/lang/words/"
        render={[Function]}
      />
      <Route
        id="wordList"
        path="/lang/words/wordlist"
        render={[Function]}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</BrowserRouter>
`;
